rust/src/ketama.rs

```diff
@@ -57,13 +74,9 @@ fn create_server_ranking(keys: &[&str]) -> Vec<ServerRank> {
             let md5_hash = Md5::digest(&hash_buf);
 
             for alignment in 0..POINTS_PER_HASH {
-                let value = u32::from_be_bytes([
-                    md5_hash[3 + alignment * 4],
-                    md5_hash[2 + alignment * 4],
-                    md5_hash[1 + alignment * 4],
-                    md5_hash[alignment * 4],
-                ]);
+                let value = ketama_hash(md5_hash.as_slice(), alignment);
                 ranking.push(ServerRank {
+                    server_name: key.to_string(),
```

Comment:
you are creating 160 copies of this string, which is insane.
It also balloons `size_of::<ServerRank>()` from 8 bytes to 32.
The `index` down below is also completely wrong now.
When you append a new node, they would all get index 0.

And `get_slot` would return that index.
The doc comment of that fn says:
> The "slot" here is an index into the origin list of keys this pool was constructed with.
So that invariant is completely broken because of this change.

Not to mention that the interaction between `get_slot` and `get_node` is completely broken.